### PR TITLE
Proof of concept-level implementation

### PR DESCRIPTION
**Pull request by Powersource**
Let's see if we can get this to work

### CODE REVIEW

**Comment by Powersource** [comment]
So it doesn't do much so far, I don't think js runs? (csp?) But it's a start!

image

File: package.json [hunk 1]
```diff
@@ -25,14 +25,16 @@
     "workbox-build": "^4.3.1"
   },
   "devDependencies": {
+    "browserify": "^16.5.0",
```
Comment (arj03) [inline package.json:29]: Would be nice to run the browserify not as a global install. Would merge a PR if you can fix that :)

**Comment by Powersource** [comment]
So I was getting "Content Security Policy: The page’s settings blocked the loading of a resource at eval (“script-src”)." for some reason so I set the csp to

```
  "content_security_policy": "script-src 'self' 'unsafe-eval'; object-src 'self';",
```

TODO fix lol

**Comment by Powersource** [comment]
But now stuff works, to some degree at least. Tried using a peer invite but that didn't work, I got this 

```
TypeError: "msg is undefined"
    acceptInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-ui.js:68651
    openInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    v moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    connect moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2493
    client moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:1949
    connect moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2493
    t moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:1675
    v moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    v moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    openInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    getInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    openInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    acceptInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-ui.js:68649
    d moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    l moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    p moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    d moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    acceptInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    l moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    acceptInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    acceptInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-core.js:2728
    acceptInvite moz-extension://68467fe3-6d7f-4170-8003-71d2d8190a9e/js/bundle-ui.js:68645
```

@arj03 idk if that is because of the extension or was this already a bug?

image

**Comment by Powersource** [comment]
Added some more logging and the actual error was 
```
Error: "while trying to connect to:net:pub.t4l3.net:8008~shs:WndnBREUvtFVF14XYEq01icpt91753bA+nVycEJIAX4="
```

Would this even work if that pub hasn't enabled websockets (or whatever this demo uses?)? And if so, how did I manage to use a peer invite last time lol, I would guess *that* invite didn't include your server (which is the only one that's enabled websockets (or whatever it is?) or have I misunderstood stuff? i should finish watching that video you sent...), but maybe.

**Comment by arj03** [comment]
@Powersource yeah it shouldn't try to connect to net addresses. The way I have tested this is with 2 browsers. 1 for generate and 1 for consuming. It might need some work to function better with invites created from patchbay. But then again, I don't think the patchbay one has any web sockets in there, so it's probably useless. Anyway, it shouldn't die because of that ;)

**Comment by arj03** [comment]
If you want some data it is probably easier to use the blob from: https://github.com/arj03/ssb-browser-demo#onboarding-file. Or just go to a profile (top right), like you main, and from there you should be able to download the latest messages and the profile info :)

**Comment by Powersource** [comment]
Holy shit it works!

image

**Comment by Powersource** [comment]
> The way I have tested this is with 2 browsers. 1 for generate and 1 for consuming

Just tried this and I think I got a similar error as before. And then do you have a special case that allows anyone to get invited through the pub?

**Comment by Powersource** [comment]
More developments https://github.com/Powersource/ssb-webextension-demo/issues/2

**Comment by arj03** [comment]
Jävla bra @Powersource!

**Pull request merged**